src/mame/igs/igs_m027.cpp

```diff
@@ -323,6 +369,59 @@ TIMER_DEVICE_CALLBACK_MEMBER(igs_m027_state::interrupt)
 		m_maincpu->pulse_input_line(ARM7_FIRQ_LINE, m_maincpu->minimum_quantum_time()); // vbl?
 }
 
+u8 igs_m027_state::ppi_porta_r()
+{
+	logerror("%s: ppi_porta_r with m_dsw_io_select as %08x\n", machine().describe_context(), m_dsw_io_select);
+
+	switch (m_dsw_io_select & 0x7) // 0x10 is set in test mode
+	{
+	case 0x03: return m_dsw3->read();
+	case 0x05: return m_dsw2->read();
+	case 0x06: return m_dsw1->read();
+
+	default: // slqz3 does't write the io_select, maybe different PCB?
+		logerror("(unhandled)\n");
+		return 0xff;
+	}
```

Comment:
This is a switch matrix – bits 0, 1 and 2 are the row drive for DIP switch banks 1, 2 and 3, respectively.  Please do it properly by testing bits and ANDing the port values rather than using a switch statement.

Reply:
I'm not sure what anything is for now, I'm learning / experimenting and making notes, nitpicking is not appreciated

Reply:
If you’re experimenting and don’t want your code reviewed, please mark the pull request as a draft and let us know you don’t want it reviewed until you mark it ready for review.

Reply:
I still don't think you get MAME at all, everything is an iterative process.  I'm just deleting the code you don't want and leaving it with less functionality, you can add it to your own preference then.

Reply:
You don’t seem to understand working with people at all.

If you create an open pull request, it will get reviewed.  If you’re experimenting and don’t want feedback, don’t create an open pull request.  Since some people open draft pull requests for the purpose of soliciting feedback, it’s best to say you don’t want feedback on a draft pull request if that’s the case.

I keep finding obvious cases where a switch matrix is misidentified as a multiplexer or priority encoder.  The code is simpler reflects the hardware better if it’s implemented as a switch matrix.

Calling any feedback “nitpicking” and pushing commits with passive-aggressive or profanity-laced and abusive messages is not helpful.  It’s also incredibly hypocritical for someone who accuses others of “rudeness” for being direct with feedback.

Reply:
well, you can implement it then, if you're so sure that's what it is.

I was still looking around trying to make notes on what I was seeing.